src/rust/zh.dmzj/src/lib.rs

```diff
@@ -0,0 +1,414 @@
+        }
+    } else {
+        let mut filters_query: String = String::from("0");
+        if filters_list.len() != 0 {
+            for i in filters_list {
+                filters_query.push_str(&helper::i32_to_string(i));
+                filters_query.push('-');
+            }
+            filters_query.pop();
+        }
+
+        let url = format!(
+            "{}/classify/{}/{}/{}.json",
+            V3_API_URL,
+            filters_query,
+            helper::i32_to_string(sort),
+            helper::i32_to_string(page)
+        );
+
+        let data = Request::new(&url, HttpMethod::Get).json().as_array()?;
+
+        for it in data {
+            let it = it.as_object()?;
+            let id = helper::i32_to_string(it.get("id").as_int()? as i32);
+            let author = it.get("authors").as_string()?.read();
+            let title = it.get("title").as_string()?.read();
+            let cover = it.get("cover").as_string()?.read();
+            let status = match it.get("status").as_string()?.read().as_str() {
+                "连载中" => MangaStatus::Ongoing,
+                "已完结" => MangaStatus::Completed,
+                _ => MangaStatus::Unknown,
+            };
+            manga_arr.push(Manga {
+                id,
+                cover,
+                title,
+                author: String::new(),
+                artist: author,
+                description: String::new(),
+                url: String::new(),
+                categories: Vec::new(),
+                status: status,
+                nsfw: MangaContentRating::Safe,
+                viewer: MangaViewer::Rtl,
+            });
+        }
+    }
+
+    let len = manga_arr.len();
+    Ok(MangaPageResult {
+        manga: manga_arr,
+        has_more: if is_keyword { false } else { len != 0 },
+    })
+}
+
+#[get_manga_details]
+fn get_manga_details(id: String) -> Result<Manga> {
+    let url = format!(
+        "{}/comic/detail/{}?channel=android&version=3.0.0&timestamp={}",
+        V4_API_URL,
+        &id,
+        aidoku::std::current_date() as i64
+    );
+
+    let pb = helper::DECODE(&helper::GET(url).string());
+    if pb.errno == 0 {
+        let pbData = pb.data.unwrap();
+        return Ok(Manga {
+            id: id.clone(),
+            cover: pbData.cover,
+            title: pbData.title,
+            author: String::from(""),
+            artist: String::from(""),
+            description: pbData.description,
+            url: format!("{}/info/{}.html", BASE_URL, id),
+            categories: pbData
+                .types
+                .iter()
+                .map(|s| String::from(s.tag_name.clone()))
+                .collect(),
+            status: match pbData.status[0].tag_name.as_str() {
+                "连载中" => MangaStatus::Ongoing,
+                "已完结" => MangaStatus::Completed,
+                _ => MangaStatus::Unknown,
+            },
+            nsfw: MangaContentRating::Safe,
+            viewer: MangaViewer::Ltr,
```

Comment:
Is it alright to assume that everything on dmzj is safe and reads from left to right?

Reply:
yep, everything is safe.
and it's seems no way to get the read direction, just assuming it's left to right as chinese traditional reading direction.